include/zephyr/net/coap_client.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/net/coap.h>
+
+/**
+ * @typedef coap_client_response_cb_t
+ * @brief Type of the callback being called when there is response to the CoAP request.
+ *
+ * @param result_code Result code of the response. Negative if there was a failure in send.
+ *                    \ref enum coap_response_code for positive.
+ * @param offset Offset starting from 0 for the payload.
+ * @param payload Buffer containing the payload from the response.
```

Comment:
What if there is no payload?
Is this `NULL` or how do we know the pointer is valid.